File: src/js/core/src/Style/MetaCreator.ts
import type { StyleDefinitionMetaCreator } from '../StyleDefinition.js';

/**
 * Read-only view over the `meta.creator` block of a style definition.
 */
export class MetaCreator {
  #data: StyleDefinitionMetaCreator;

  constructor(data: StyleDefinitionMetaCreator) {
    this.#data = data;
  }

  /**
   * Returns the creator's display name, or `undefined` when not set.
   */
  name(): string | undefined {
    return this.#data.name;
  }

  /**
   * Returns the creator's homepage URL, or `undefined` when not set.
   */
  url(): string | undefined {
    return this.#data.url;
  }
}

File: src/dart/core/lib/src/style/meta_creator.dart
/// Read-only view over the `meta.creator` block of a style definition.
library;

/// The creator descriptor of a style's metadata.
///
/// Accessors return `null` for absent fields, distinguishing them from fields
/// explicitly set to the empty string — the license builder relies on that
/// distinction (the JS nullish `?? 'Unknown'`; Go/Rust use `*string`/Option
/// for the same reason).
class MetaCreator {
  final Map<String, Object?> _data;

  MetaCreator(Map<String, Object?> data) : _data = data;

  /// Returns the creator's display name, or `null` when not set.
  String? name() => _data['name'] as String?;

  /// Returns the creator's homepage URL, or `null` when not set.
  String? url() => _data['url'] as String?;
}
